ipfs/defaults.go

```diff
@@ -0,0 +1,201 @@
+package ipfs
+
+import (
+	"os"
+
+	ipfscfg "github.com/ipfs/go-ipfs-config"
+	"github.com/ipfs/interface-go-ipfs-core/options"
+)
+
+// BootstrapPeers is a list of default bootstrap peers for IPFS network
+// TODO: Change this to LL defaults at some point
+var BootstrapPeers, _ = ipfscfg.DefaultBootstrapPeers()
+
+// DefaultFullNodeConfig provides default embedded IPFS configuration for FullNode
+func DefaultFullNodeConfig() (*ipfscfg.Config, error) {
+	identity, err := ipfscfg.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+		options.Key.Type(options.Ed25519Key),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var conf = &ipfscfg.Config{
+		// ed25519 PKI identity for p2p communication
+		// TODO: Any reason not reuse ed25519 key from consensus identity?
```

Comment:
That is a very good question. I also thought about this a lot. I think the main reason for now is that @adlerjohn wants to use secp256k1 instead of ed25519 for consensus signing too. AFAIR secp is not supported by ipfs.
I'm back and fort on this. I do not care too much about secp vs ed25519 tbh but the general question if we should have one key for all pupose or several keys for different purposes. It might be not a bad idea to make it non-trivial to discover which validator runs which ipfs node for instance.

Reply:
@liamsi, libp2p support secp256k1 so we can use it https://github.com/libp2p/go-libp2p-core/blob/master/crypto/secp256k1.go

Reply:
I think TODO(@name) should have a name of a contributor that this todo is addressed, not the name of one who added it. So your suggestions do not make sense to me, as I am not the one who this question is addressed

Reply:
I will add support for secp256k1 keys in follow up PR

Reply:
> So your suggestions do not make sense to me, as I am not the one who this question is addressed

Todos in the code are not the right place to ask questions.

Reply:
> libp2p supports secp256k1 so we can use it

Nice!

Reply:
@Wondertan want to open a pr then I can remove the ed25519 key types from the repo. Less dependencies

Reply:
@marbar3778, don't get what that PR should do. Do you mean a PR that changes ed25519 to secp256k1 in IPFS?